demos/common/python/models/bert.py

```diff
@@ -0,0 +1,196 @@
+"""
+ Copyright (c) 2021 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+import numpy as np
+
+from .model import Model
+
+
+class Bert(Model):
+
+    def __init__(self, ie, model_path, vocab):
+        super().__init__(ie, model_path)
+        self.token_cls = [vocab['[CLS]']]
+        self.token_sep = [vocab['[SEP]']]
+        self.token_pad = [vocab['[PAD]']]
+
+    def check_input_output_names(self, input_names, output_names=None):
+        self.input_names = [i.strip() for i in input_names.split(',')]
+        if self.net.input_info.keys() != set(self.input_names):
+            raise RuntimeError('The demo expects input names: {}, actual network input names: {}'.format(
+                self.input_names, list(self.net.input_info.keys())))
+
+        self.max_length = self.net.input_info[self.input_names[0]].input_data.shape[1]
+
+        if output_names is None:
+            return
+
+        self.output_names = [o.strip() for o in output_names.split(',')]
+        if self.net.outputs.keys() != set(self.output_names):
+            raise RuntimeError('The demo expects output names: {}, actual network output names: {}'.format(
+                self.output_names, list(self.net.outputs.keys())))
+
+    def preprocess(self, inputs):
+        input_ids, attention_mask, token_type_ids = self.form_request(inputs)
+
+        pad_len = self.pad_input(input_ids, attention_mask, token_type_ids)
+        meta = {'pad_len': pad_len, 'inputs': inputs}
+
+        return self.create_input_array(input_ids, attention_mask, token_type_ids), meta
+
+    def pad_input(self, input_ids, attention_mask, token_type_ids):
+        pad_len = self.max_length - len(input_ids)
+        if pad_len < 0:
+            raise ValueError("The input request is longer than max number of tokens ({})"
+                             " processed by model".format(self.max_length))
+        input_ids += self.token_pad * pad_len
+        token_type_ids += self.token_pad * pad_len
+        attention_mask += self.token_pad * pad_len
```

Comment:
token_type_ids and attention_mask should be padded with `0`s, not with token_pad id. For some vocabs, it may not be `0`.

Reply:
Thank you for the clarification, updated.